pkg/kubectl/resource_printer.go

```diff
@@ -281,6 +287,16 @@ type YAMLPrinter struct {
 
 // PrintObj prints the data as YAML.
 func (p *YAMLPrinter) PrintObj(obj runtime.Object, w io.Writer) error {
+	switch obj := obj.(type) {
+	case *runtime.Unknown:
+		data, err := yaml.JSONToYAML(obj.RawJSON)
```

Comment:
Similar marshalling question here.  I understand that the additional step shouldn't be required, but having to special case it is likely to lead to issues in other code paths.